datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +50,26 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
```

Comment:
One good validation will be to check if `assignedPartitions` and `unassignedPartitions` have any intersection. This is critical as we don't want two tasks to have the same partition.

Reply:
This will be done in the strategy class itself.